File: goApi/app/middleware/crossOrigin.go
package middleware

import (
	"github.com/gin-gonic/gin"
	"net/http"
)

func Cors() gin.HandlerFunc {
	return func(c *gin.Context) {
		method := c.Request.Method

		c.Header("Access-Control-Allow-Origin", "*")
		c.Header("Access-Control-Allow-Methods", "GET,POST,PATCH,PUT,DELETE,OPTIONS,DELETE")
		c.Header("Access-Control-Allow-Credentials", "true")

		c.Header("Access-Control-Allow-Headers", "Authori-zation,Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since, X-Requested-With, AccessToken,X-CSRF-Token, Authorization, Token, token")

		c.Header("Access-Control-Max-Age", "1728000")
		c.Header("Content-Type", "application/json; charset=utf-8")
		c.Header("Access-Control-Expose-Headers", "Content-Length, Access-Control-Allow-Origin, Access-Control-Allow-Headers, Content-Type")







		//放行所有OPTIONS方法
		if method == "OPTIONS" {
			c.AbortWithStatus(http.StatusNoContent)
		}

		//处理请求
		c.Next()

	}
}

File: phpFileSys/App/Bootstrap/CrossOrigin.php
<?php


namespace App\Bootstrap;;
use EasySwoole\Component\Di;

class CrossOrigin
{
    public static function boot(){
        //配置跨域
        // onRequest v3.4.x+
        Di::getInstance()->set(\EasySwoole\EasySwoole\SysConst::HTTP_GLOBAL_ON_REQUEST, function (\EasySwoole\Http\Request $request, \EasySwoole\Http\Response $response) {
            $response->withHeader("Access-Control-Allow-Origin", "*");
            $response->withHeader("Access-Control-Allow-Methods", 'GET, POST, DELETE, PUT, PATCH, OPTIONS');
            $response->withHeader("Access-Control-Allow-Credentials", "true");
            $response->withHeader("Access-Control-Allow-Headers", "Authorization, User-Agent, Keep-Alive, Content-Type, X-Requested-With, Authori-zation, Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since, X-Requested-With, AccessToken, X-CSRF-Token, Token, token");
            $response->withHeader("Access-Control-Max-Age", "1728000");
            $response->withHeader("Content-Type", "application/json; charset=utf-8");
            if ($request->getMethod() === 'OPTIONS') {
                file_put_contents(EASYSWOOLE_ROOT.'/Temp/liaolog',print_r($request->getMethod(),true),FILE_APPEND);
                $response->withStatus(http_response_code());
                $response->end();
                return  false;   //务必返回 false，断掉 中间件的后续操作的意思？
            }
            return true;
        });
    }
}
